server/errors/errors.go

```diff
@@ -44,6 +44,12 @@ var (
 		Description:    "An error occurred when attempting to apply an update at the storage layer.",
 		HTTPStatusCode: http.StatusInternalServerError,
 	})
+	ErrOldVersion = errcode.Register(errGroup, errcode.ErrorDescriptor{
+		Value:          "VERSION",
+		Message:        "A newer version of metadata is already available.",
+		Description:    "A newer version of the repository's metadata is already available in storage.",
+		HTTPStatusCode: http.StatusInternalServerError,
```

Comment:
It's more likely this will be raised because a user sends us an old version. Snapshot generation means it could technically be an internal server error, but the majority case is expected to be bad user input so `http.StatusBadRequest` is probably more appropriate.